api/redirects.go

```diff
@@ -62,3 +72,123 @@ func isValidBase64(s string) bool {
 	_, err := base64.StdEncoding.DecodeString(s)
 	return err == nil
 }
+
+// encodeBase64 returns the base64 encoded string of the original URL key string
+func encodeBase64(key string) string {
+	encodedKey := base64.StdEncoding.EncodeToString([]byte(key))
+
+	return encodedKey
+}
+
+// getRedirects gets a paged list of redirects from the store
+func (api *RedirectAPI) getRedirects(w http.ResponseWriter, req *http.Request) {
+	ctx := req.Context()
+	strCount := req.URL.Query().Get("count")
+	strCursor := req.URL.Query().Get("cursor")
+
+	// make count default to 10
+	if strCount == "" {
+		strCount = "10"
+	}
+
+	// make cursor default to 0
+	if strCursor == "" {
+		strCursor = "0"
+	}
+
+	logData := log.Data{"count": strCount, "cursor": strCursor}
+
+	// convert count from a string to an int64 ready to use in call to GetRedirects
+	count, errCount := strconv.ParseInt(strCount, 10, 32)
+
+	// validate count
+	if count < 0 {
+		errNegCount := errors.New("the count is negative")
+		log.Error(ctx, "invalid path parameter - count should be a positive integer", errNegCount, logData)
+		api.handleError(ctx, w, apierrors.ErrNegativeCount, http.StatusBadRequest)
+		return
+	}
+
+	if errCount != nil {
+		log.Error(ctx, "invalid path parameter - failed to convert count to int64", errCount, logData)
+		api.handleError(ctx, w, apierrors.ErrInvalidCount, http.StatusBadRequest)
+		return
+	}
+
+	// convert cursor from a string to a uint64 ready to use in call to GetRedirects
+	var cursor uint64
+	var errCursor error
+	cursor, errCursor = strconv.ParseUint(strCursor, 10, 32)
+
+	logData = log.Data{"count": count, "cursor": cursor}
+
+	// validate cursor
+	if errCursor != nil {
+		log.Error(ctx, "invalid path parameter - failed to convert cursor to uint64", errCursor, logData)
+		api.handleError(ctx, w, apierrors.ErrInvalidOrNegativeCursor, http.StatusBadRequest)
+		return
+	}
+
+	keyValuePairs, newCursor, errRedirects := api.RedirectStore.GetRedirects(ctx, count, cursor)
+	if errRedirects != nil {
+		log.Error(ctx, "error calling the store to get redirects", errRedirects, logData)
+		api.handleError(ctx, w, apierrors.ErrRedis, http.StatusInternalServerError)
+		return
+	}
+
+	logData = log.Data{"numKeyValuePairs": len(keyValuePairs)}
+	log.Info(ctx, "The number of redirects fetched", logData)
+
+	redirectBase := "https://api.beta.ons.gov.uk/v1/redirects/"
```

Comment:
Don't hardcode the HATEOS links. We have some links middleware in dp-net to help get this from the request headers. See for example https://github.com/ONSdigital/dp-image-api/pull/101

Reply:
👍